kernel/src/checkpoint/mod.rs

```diff
@@ -391,3 +459,42 @@ fn deleted_file_retention_timestamp_with_time(
     // Simple subtraction - will produce negative values if retention > now
     Ok(now_ms - retention_ms)
 }
+
+/// Creates the data for the _last_checkpoint file containing checkpoint
+/// metadata with the `create_one` method. Factored out to facilitate testing.
+///
+/// # Parameters
+/// - `engine`: Engine for data processing
+/// - `version`: Table version number
+/// - `actions_counter`: Total actions count
+/// - `add_actions_counter`: Add actions count
+/// - `size_in_bytes`: Size of the checkpoint file in bytes
+///
+/// # Returns
+/// A new [`EngineData`] batch with the `_last_checkpoint` fields:
+/// - `version` (i64, required): Table version number
+/// - `size` (i64, required): Total actions count
+/// - `parts` (i64, optional): Always 1 for single-file checkpoints
+/// - `sizeInBytes` (i64, optional): Size of checkpoint file in bytes
+/// - `numOfAddFiles` (i64, optional): Number of Add actions
+///
+/// TODO(#838) Add `checksum` field to `_last_checkpoint` file
+/// TODO(#839) Add `checkpoint_schema` field to `_last_checkpoint` file
+pub(crate) fn create_last_checkpoint_data(
+    engine: &dyn Engine,
+    version: i64,
+    actions_counter: i64,
+    add_actions_counter: i64,
+    size_in_bytes: i64,
+) -> DeltaResult<Box<dyn EngineData>> {
+    engine.evaluation_handler().create_one(
+        LAST_CHECKPOINT_SCHEMA.clone(),
+        &[
+            version.into(),
+            actions_counter.into(),
+            1i64.into(),
```

Comment:
maybe useful to also add quick inline note for the const (i know listed above but just for ease)
```suggestion
            1i64.into(), // parts = 1 since we only support single-part checkpoint here
```